velero-plugins/imagestream/shared_test.go

```diff
@@ -0,0 +1,423 @@
+						APIVersion: "v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "ns1",
+					},
+				},
+				&corev1.Secret{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Secret",
+						APIVersion: "v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "oadp-bsl1-aws-registry-secret",
+						Namespace: "ns1",
+					},
+					Data: map[string][]byte{
+						"access_key": []byte("ak"),
+						"secret_key": []byte("sk"),
+					},
+				},
+				&velerov1.BackupStorageLocation{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "BackupStorageLocation",
+						APIVersion: velerov1.SchemeGroupVersion.String(),
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "bsl1",
+						Namespace: "ns1",
+					},
+					Spec: velerov1.BackupStorageLocationSpec{
+						Provider: "aws",
+						Default:  true,
+						StorageType: velerov1.StorageType{
+							ObjectStorage: &velerov1.ObjectStorageLocation{
+								Bucket: "buc",
+								Prefix: "prefix",
+							},
+						},
+						Credential: &corev1.SecretKeySelector{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: "cloud-credentials",
+							},
+							Key: "cloud",
+						},
+					},
+				},
+			},
+			want: []string{
+				"REGISTRY_STORAGE=s3",
+				"REGISTRY_STORAGE_S3_BUCKET=buc",
+				"REGISTRY_STORAGE_S3_REGION=us-east-2",
+				"REGISTRY_STORAGE_S3_ACCESSKEY=ak",
+				"REGISTRY_STORAGE_S3_SECRETKEY=sk",
+			},
+		},
+		{
+			name: "aws case, operator did not create registry secret",
+			args: args{
+				location:  "bsl1",
+				namespace: "ns1",
+			},
+			objs: []runtime.Object{
+				&corev1.Namespace{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Namespace",
+						APIVersion: "v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "ns1",
+					},
+				},
+				&velerov1.BackupStorageLocation{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "BackupStorageLocation",
+						APIVersion: velerov1.SchemeGroupVersion.String(),
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "bsl1",
+						Namespace: "ns1",
+					},
+					Spec: velerov1.BackupStorageLocationSpec{
+						Provider: "aws",
+						Default:  true,
+						StorageType: velerov1.StorageType{
+							ObjectStorage: &velerov1.ObjectStorageLocation{
+								Bucket: "buc",
+								Prefix: "prefix",
+							},
+						},
+						Credential: &corev1.SecretKeySelector{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: "cloud-credentials",
+							},
+							Key: "cloud",
+						},
+					},
+				},
+			},
+			want:          []string{},
+			wantErr:       true,
+			wantErrString: "secrets \"oadp-bsl1-aws-registry-secret\" not found",
+		},
+	}
+	scheme := runtime.NewScheme()
+	corev1.AddToScheme(scheme)
+	metav1.AddMetaToScheme(scheme)
+	velerov1.AddToScheme(scheme)
+	var bslCRD apiextensionsv1.CustomResourceDefinition
+	err := yaml.Unmarshal([]byte(bslCRDyaml), &bslCRD)
+	if err != nil {
+		panic(err)
+	}
+	testEnv := &envtest.Environment{
```

Comment:
This is the third place that `envtest` is being initialized

https://github.com/openshift/openshift-velero-plugin/blob/628f014a28f70d8c307c44ec08b529e4c1c4995a/velero-plugins/build/restore_test.go

https://github.com/openshift/openshift-velero-plugin/blob/628f014a28f70d8c307c44ec08b529e4c1c4995a/velero-plugins/imagestream/registry_test.go

we should have a centralized startup/cleanup for it, like in OADP. Example https://github.com/migtools/oadp-non-admin/blob/master/internal/controller/suite_test.go

Reply:
Unit tests are run separately. Just consolidating code? This envtest add crd where others do not.

Do you just want a parameterized envtestFunc?

Reply:
Linked code is for ginkgo ewe.. not unit test.
This repo does not have centralized suite e2e

Reply:
It could perhaps be done if all the tests requiring envtest is in the same package so it can run together as package tests.